src/java/org/apache/cassandra/index/sai/disk/v1/IndexSearcher.java

```diff
@@ -104,12 +103,13 @@ protected IndexSearcher(PrimaryKeyMap.Factory primaryKeyMapFactory,
      * Order the on-disk index synchronously and produce an iterator in score order
      *
      * @param orderer      the object containing the ordering logic
+     * @param filter       optional predicate to get a slice of the index
      * @param keyRange     key range specific in read command, used by ANN index
      * @param queryContext to track per sstable cache and per query metrics
      * @param limit        the num of rows to returned, used by ANN index
      * @return an iterator of {@link PrimaryKeyWithSortKey} in score order
      */
-    public abstract CloseableIterator<PrimaryKeyWithSortKey> orderBy(Orderer orderer, AbstractBounds<PartitionPosition> keyRange, QueryContext queryContext, int limit) throws IOException;
+    public abstract CloseableIterator<PrimaryKeyWithSortKey> orderBy(Orderer orderer, Expression filter, AbstractBounds<PartitionPosition> keyRange, QueryContext queryContext, int limit) throws IOException;
```

Comment:
Nit: what do you think about calling it a `slice` instead of a `filter`? I can see `filter` because it is generic, but I also don't want it to imply that we're post filtering results. Another option could be `termRange` or `termSlice` to be more specific. I don't feel strongly, just looking to collaborate.

Reply:
Makes a lot more sense. Slice is a better name! Thanks.